application/src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -360,24 +361,15 @@ private List<Leg> mapNonTransitLeg(
           .build()
       );
     } else {
-      StateEditor se = new StateEditor(edges.get(0).getFromVertex(), transferStreetRequest);
-      se.setTimeSeconds(createZonedDateTime(pathLeg.fromTime()).toEpochSecond());
-
-      State s = se.makeState();
-      ArrayList<State> transferStates = new ArrayList<>();
-      transferStates.add(s);
-      for (Edge e : edges) {
-        var states = e.traverse(s);
-        if (State.isEmpty(states)) {
-          s = null;
-        } else {
-          transferStates.add(states[0]);
-          s = states[0];
-        }
-      }
-
-      State[] states = transferStates.toArray(new State[0]);
-      var graphPath = new GraphPath<>(states[states.length - 1]);
+      var legTransferSearchRequest = transferStreetRequest
```

Comment:
This one didn't use the EdgeTraverser before. What is the reason for using it here?

Reply:
Multiple states may also occur here, which need to be handled in a similar manner:
* there could be multiple initial states
* the edge traversal could return multiple states, of which the first one may not be the optimal

As an example, the `CAR_PICKUP` transfers resulted in an NPE, since the initial `CAR/IN_CAR` state couldn't traverse the `StreetTransitLink` entities.

By using `EdgeTraverser` traversal results in the same state as when creating the actual transfer.